cpp/visualization/serialization.cpp

```diff
@@ -0,0 +1,221 @@
+#include "serialization.hpp"
+#include "helpers.hpp"
+
+#include <iostream>
+#include <nlohmann/json.hpp>
+
+namespace spectacularAI {
+namespace visualization {
+namespace {
+
+// Deleting directory in C++ without <filesystem> is painful...
+// Use python instead...
+const char* pythonScript =
```

Comment:
Probably most ugly thing in this PR. But somewhat OK, because we already use this pattern in `visualization.cpp`.